### ISSUE TITLE
Value Error when run InaSAFE analysis for Flood Hazard

### ISSUE THREAD

**Issue by adissadis** [AIFDR/inasafe#1728]
# Problem
I encountered this error message when i try to analysis AsiaPop (Raster) with Flood hazard from BPBD DKI Jakarta (Vector). 

```
Problem

The following problem(s) were encountered whilst running the analysis.

ValueError

Suggestion

You can try the following to resolve the issue:

Check that you have the latest version of InaSAFE installed - you may have encountered a bug that is fixed in a subsequent release.
Check the InaSAFE documentation to see if you are trying to do something unsupported.
Report the problem using the issue tracker at https://github.com/AIFDR/inasafe/issues. Reporting an issue requires that you first create a free account at http://github.com. When you report the issue, please copy and paste the complete contents of this panel into the issue to ensure the best possible chance of getting your issue resolved.
Try contacting one of the InaSAFE development team by sending an email to info@inasafe.org. Please ensure that you copy and paste the complete contents of this panel into the email.
Details

These additional details were reported when the problem occurred.

An exception occurred when calculating the results. Calculation error encountered:
cannot convert float NaN to integer
Diagnostics (click for details)
```

```python
In file "C:/Users/Sony/.qgis2/python\inasafe\safe\utilities\impact_calculator_thread.py", line 189, in run check_integrity=self._check_integrity)
In file "C:/Users/Sony/.qgis2/python\inasafe\safe\engine\core.py", line 71, in calculate_impact F = impact_function.run(layers)
In file "C:/Users/Sony/.qgis2/python\inasafe\safe\impact_functions\inundation\flood_population_evacuation_polygon_hazard.py", line 294, in run affected_population)
In file "C:/Users/Sony/.qgis2/python\inasafe\safe\impact_functions\core.py", line 156, in population_rounding_full number = int(rounding * ceil(1.0 * number / rounding))
```

# Platform
Windows 7 64 bit QGIS 2.8.1 InaSAFE 3.1.0 dev-master

**Comment by timlinux** [comment]
@cchristelis note this fix should be backported to master too

**Comment by cchristelis** [comment]
Changing the code to not produce this error can be done in a number of different ways. To select the best method we will have to look at what the underlying reason is for the nan. This is a property of the raster layer. I was able to reproduce this issue by adding a nan layer as population. 

1) The first way in which this problem could be solved is by handling nan's in various places in the code, ignoring them. This is equivalent to assuming "the population was 0" and could also be handled as such.

2) The next option we have is to inform the user that there is a problem with the input data and the data should be corrected before the analysis is run.

3) Finally we have the option of assigning a sensible value to the missing pixel. This option presents several sub-options, where neighboring pixels are used to fill this value.

Choosing a method to solve this issue requires some GIS insight and hence I would like to know what you think @timlinux  and @adissadis?

**Comment by cchristelis** [comment]
After chatting with Tim about this, it became clear that for vector on raster operations, such as this example the rules of Zonal stats should be applied, which effectively means that the Null/NaN values are not contributing to the output (treated as 0, which was the first option.) For raster on raster operations, the Null/Nan values should be propagated.

**Comment by akbargumbira** [comment]
For this specific analysis (vector on raster), all the IF's will treat the nodata values as 0 to count the total affected and total population. I also added a statement telling that in the Notes section. I am closing this particular issue then. The higher handling to inform the user if they have nodata values will be addressed on the other ticket by @cchristelis #1800
